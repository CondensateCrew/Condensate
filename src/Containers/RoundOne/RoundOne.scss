@import "../../css/variables.scss";

.round-one-board {
  display: grid;
  grid-template-rows: 1fr 8fr 1fr;
  height: 100vh;
  overflow: hidden;
  padding: $medium-space;
  footer {
    align-self: end;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
  }
}
