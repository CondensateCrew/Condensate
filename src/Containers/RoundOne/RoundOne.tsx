import './RoundOne.scss';
import React from 'react';
import Header from 'Components/Header/Header';
import GameBoard from 'Containers/GameBoard/GameBoard';
import ChosenWordTrain from 'Containers/ChosenWordTrain/ChosenWordTrain';
import Timer from 'Components/Timer/Timer';

const RoundOne:React.FC = () => {
  return (
    <main className="round-one-board">
      <Header />
      <GameBoard />
      <footer>
        <ChosenWordTrain />
        <Timer time={90} />
      </footer>
    </main>
  )
}

export default RoundOne;
