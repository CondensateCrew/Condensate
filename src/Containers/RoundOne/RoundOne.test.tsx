import React from 'react';
import RoundOne from './RoundOne';
import { shallow } from 'enzyme';
import { useSelector } from 'react-redux';
jest.mock('react-redux');

describe('RoundOne component', () => {
  it('should match the snapshot', () => {
    useSelector.mockImplementation(() => false);
    const wrapper = shallow(<RoundOne />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if time is ended', () => {
    useSelector.mockImplementation(() => true);
    const wrapper = shallow(<RoundOne />);
    expect(wrapper).toMatchSnapshot();
  });
});
