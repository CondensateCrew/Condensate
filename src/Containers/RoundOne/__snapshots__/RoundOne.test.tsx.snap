// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundOne component should match the snapshot 1`] = `
<main
  className="round-one-board"
>
  <Header />
  <GameBoard />
  <footer>
    <ChosenWordTrain />
    <Timer
      time={60}
    />
  </footer>
</main>
`;

exports[`RoundOne component should match the snapshot if time is ended 1`] = `
<main
  className="round-one-board"
>
  <Header />
  <GameBoard />
</main>
`;
